### DISCUSSION TITLE
pdfcpu merge -m create vs. pdfcpu merge -m append

### DISCUSSION THREAD

**Discussion by johnmapeson** [pdfcpu/pdfcpu#927] [Q&A] [answered]
I want to concatenate two PDF files, the first contains pages 1-100 and the second pages 101-200. Do I need to use `pdfcpu merge output.pdf input1.pdf input2.pdf` or instead `pdfcpu merge -m append output.pdf input1.pdf input2.pdf`?

I do understand that both the commands work and (at least in most cases) produce the same output . But when working with other PDF processors, I noticed that when concatenating documents, it is generallay safer to append one document to another rather then to create a new one: in the latter case, bookmarks, layers, hyperlinks are sometimes missing or broken.

Does the same recommendation, the recommendation to append one document to another rather then to create a new one when concatenating them, is relevant for pdfcpu?

**Comment by hhrutter** [comment, marked as answer]
Technically there is not much difference between merging in `create` or `append` mode so comes down to a matter of taste.
So you can do it either way. It's all about easy to follow semantics.

You always have one input file and merge the remaining input files one by one onto the first one.

In `create`  mode pdfcpu writes back the merge result into a brand new file.

In `append' mode pdfcpu will merge all input files into the first one if it already exists otherwise the behavior falls back to `create`.

Bookmarks are preserved across the board and other major PDF features are preserved.
Struct Trees are definitely not yet merged/carried over.

If you stumble across a particular merge scenario where you miss smth, please open an issue and we follow up.
